reactive/containerd.py

```diff
@@ -378,11 +414,14 @@ def config_changed():
         context
     )
 
+    DB.set('config-cache', context)
     set_state('containerd.restart')
 
 
 @when('containerd.ready')
```

Comment:
```suggestion
@when('containerd.installed')
```

I think `.ready` here was the reason this bug cropped up in the first place.  See my [bug commentary](https://bugs.launchpad.net/charm-containerd/+bug/1874910/comments/8) for details. TL;DR is that we may not be `.ready` when the `config-changed` hook runs.  Feels safer to gate this on `.installed` instead.